src/backend/gpopt/utils/COptTasks.cpp

```diff
@@ -986,14 +986,15 @@ COptTasks::PrintMissingStatsWarning(CMemoryPool *mp, CMDAccessor *md_accessor,
 
 	if (0 < rel_stats->Size())
 	{
-		int length = NAMEDATALEN * rel_stats->Size() + 200;
-		char msgbuf[length];
-		snprintf(
-			msgbuf, sizeof(msgbuf),
-			"One or more columns in the following table(s) do not have statistics: %s",
+		CWStringDynamic msgbuf(mp);
+		msgbuf.AppendFormat(
+			GPOS_WSZ_LIT(
+				"One or more columns in the following table(s) do not have statistics: %s"),
```

Comment:
msgbuf is string of WCHAR and wcstr is string of WCHAR too. So we can replace "%s" with "%ls" and don't call CreateMultiByteCharStringFromWCString

Reply:
Done